### ISSUE TITLE
Vim script cannot be commented correctly with multiple quotation marks

### ISSUE THREAD

**Issue by netvolcano** [vim/vim#8768]
**Vim script cannot be commented correctly with multiple quotation marks**
I found a bug in the vim language syntax of <vim.vim> file. If there are multiple quotation marks in a comment, it will not be treated as normal comment. For example:
I wrote a vim script as below:
```
  let a = b    " b range:0~255
```
_Note: there are some spaces at the beginning of the statement._

I want to comment this statement, so add quotation at the beginning of the statement.
```
  " let a = b    " b range:0~255
```
I find the statement will NOT be recognized as `vimComment` as I expected, it will be treated as `vimString` (when cursor under let) and `vimIsCommand` (when cursor under range).

Here, I used the following method to check syntax name and definition under cursor. 
```
nnoremap <f8> :echo synIDattr(synID(line('.'), col('.'), 0), 'name')<cr>
```

**To Reproduce**
Write a vim script as the above, if you insert multiple quotations in a line.

**Expected behavior**
I expected all words in the commented lines will be vimComment .

**Screenshots**
https://i.stack.imgur.com/oBmif.png

**Environment (please complete the following information):**
 - Vim version [8.2 2019 Dec 12.]
 - OS: [Windows 10 1809]
 - Vim.vim version [8.0-28]

**Comment by brammool** [comment]
I do not see this problem.  Please get the latest syntax file version, should be 8.2-06.

**Comment by netvolcano** [comment]
> 
> 
> I do not see this problem. Please get the latest syntax file version, should be 8.2-06.

I tried the lastest syntax version and found the same bug too. Please check you test vim script and make sure that there are some spaces at the begging of the statement.
```
----" let a = b   " b range:0~255
```

Note: the '----' in the above example represents the space characters.

I found if there are no spaces at the begginng, there will be not this problem.

**Comment by netvolcano** [comment]
I used the default .vimrc and without any plugins, there will be not this problem. I think it is caused by some plugins. How can find which plugins will determine the syntax of a line?

**Comment by cecamp** [comment]
The item looks like a string surrounded by double-quotes and is being highlighted as such. Looks difficult to have syntax/vim.vim discern the difference as a double-quote character essentially has overloaded meaning to it (comment vs starting a string). The problem is one of lexical vs syntactic analysis; vim does provide some syntactic analysis (contains, nextgroup, etc) but is liable to drastically increasing the size of the syntax file. Note that changing the b to bb also changes its highlighting, as b is a command (shorthand for buffer) but bb is just a variable.

With vim 8.2.x comes a new commenting character, the "#". I advise trying that to comment out blocks of vim script.

**Comment by brammool** [comment]
Yeah, the old Vi way for comments can be confusing.  However, the example with leading space and then a double quote works fine for me, I suspect you still use the old syntax file.

**Comment by netvolcano** [comment]
> The item looks like a string surrounded by double-quotes and is being highlighted as such. Looks difficult to have syntax/vim.vim discern the difference as a double-quote character essentially has overloaded meaning to it (comment vs starting a string). The problem is one of lexical vs syntactic analysis; vim does provide some syntactic analysis (contains, nextgroup, etc) but is liable to drastically increasing the size of the syntax file. Note that changing the b to bb also changes its highlighting, as b is a command (shorthand for buffer) but bb is just a variable.
> 
> With vim 8.2.x comes a new commenting character, the "#". I advise trying that to comment out blocks of vim script.

I think if the `vimString` and `vimIsCommand` will NOT be contained in the `vimLineComment` , the issue will be avoid. Is there any method to implement this?

**Comment by netvolcano** [comment]
For received feedbacks that there were no the same issue found in the others' VIM enviroment, I had to check my settings and did some trials. In the end I found this issue was caused by the `Plugin 'thaerkh/vim-indentguides'`. If I did not use this plugin, the comment state was correct as I expected. I guess the plugin has some errors in using of VIM conceal feature. So I changed to use the `Plugin 'Yggdroot/indentLine'` instead and solved the issue. Thank @all.
